### PR TITLE
support autoincrement PKs in dm_rows_append()

### LINKED ISSUE #1386: dm_rows_append() with IDENTITY columns
doesn't work today, columns need to be remapped. We could:

- [x] #1642
- [ ] for such PKs, we would call `rows_append(returning = )` and remap ID columns in the child tables

### LINKED ISSUE #1695: test for `tbl_sql` instead of `tbl_dbi`?
In `dm_validate()`, should we test for inheritance of `data.frame` and `tbl_sql` instead of `data.frame` and `tbl_dbi`?
Most of the times it works, but it doesn't, if we create a DB table with `copy_inline()`:
``` r
library(dm)

con <- DBI::dbConnect(RPostgres::Postgres())
test_table_pg <- dbplyr::copy_inline(con, tibble(a = 1:3))
class(test_table_pg)
#> [1] "tbl_PqConnection" "tbl_sql"          "tbl_lazy"         "tbl"
dm(test_table_pg)
#> Error in `abort_dm_invalid()` at dm/R/validate.R:37:4:
#> ! This `dm` is invalid, reason: Not all entries in `def$data` are of class `data.frame` or `tbl_dbi`. Check `dm_get_tables()`.
```

<sup>Created on 2022-12-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### PR DESCRIPTION
**Pull request by TSchiefer**
on Postgres, SQL Server, SQLite

closes #1695
closes #1386

### CODE REVIEW

**Comment by TSchiefer** [comment]
I think 1b7af4f7c036c7dfc4304b2bc0eeee9e2cbd08b0 is responsible for the failing validate-GHA?

**Comment by krlmlr** [comment]
Good catch, I tweaked main, let's see if validate passes there.

**Comment by krlmlr** [comment]
Fixed workflow failures, let's see.

I'll review the code too.

File: R/rows-dm.R [hunk 1]
```diff
@@ -287,7 +287,7 @@ dm_rows_run <- function(x, y, rows_op_name, top_down, in_place, require_keys, pr
     if (!(all(tables %in% all_pks$table))) {
       abort(glue("`dm_rows_{rows_op_name}()` requires the 'dm' object to have primary keys for all target tables."))
     }
-    keys <- deframe(dm_get_all_pks(x))[tables]
+    keys <- deframe(select(all_pks, table, pk_col))[tables]
```
Comment (krlmlr) [inline R/rows-dm.R:290]: Something more base-y?

```suggestion
    keys <- all_pks$pk_col[match(tables, all_pks$table)]
```

**Review by krlmlr — COMMENTED** [review]
Thanks! I'll tweak and submit later today, fingers crossed.

**Comment by krlmlr** [comment]
WAT is this Windows R 3.6 failure???

**Comment by krlmlr** [comment]
It's also present for the main branch: https://github.com/cynkra/dm/pull/1703. I'll take a look.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**